consensus/quorum/one-node-staked-vote.go

```diff
@@ -1,75 +1,177 @@
 package quorum
 
 import (
+	"encoding/json"
 	"math/big"
 
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/harmony-one/bls/ffi/go/bls"
+	"github.com/harmony-one/harmony/internal/utils"
 	"github.com/harmony-one/harmony/numeric"
 	"github.com/harmony-one/harmony/shard"
+	"github.com/harmony-one/harmony/staking/slash"
 )
 
 var (
-	twoThirds = numeric.NewDec(2).QuoInt64(3).Int
+	twoThirds = numeric.NewDec(2).QuoInt64(3)
+	hSentinel = numeric.ZeroDec()
 )
 
 type stakedVoter struct {
 	isActive, isHarmonyNode bool
+	earningAccount          common.Address
 	effective               numeric.Dec
 }
 
 type stakedVoteWeight struct {
 	SignatureReader
 	DependencyInjectionWriter
-	// EPOS based staking
-	validatorStakes            map[[shard.PublicKeySizeInBytes]byte]stakedVoter
-	totalEffectiveStakedAmount *big.Int
+	DependencyInjectionReader
+	slash.ThresholdDecider
+	validatorStakes map[[shard.PublicKeySizeInBytes]byte]stakedVoter
+	total           numeric.Dec
 }
 
 // Policy ..
 func (v *stakedVoteWeight) Policy() Policy {
 	return SuperMajorityStake
 }
 
-// We must maintain 2/3 quoroum, so whatever is 2/3 staked amount,
-// we divide that out & you
 // IsQuorumAchieved ..
 func (v *stakedVoteWeight) IsQuorumAchieved(p Phase) bool {
-	// TODO Implement this logic
+	// TODO Implement this logic w/Chao
+	// soFar := numeric.ZeroDec()
+	w := shard.BlsPublicKey{}
+	members := v.Participants()
+
+	for i := range members {
+		w.FromLibBLSPublicKey(members[i])
+		// isHMY := v.validatorStakes[w].isHarmonyNode
+		if v.ReadSignature(p, members[i]) == nil {
+			//
+		}
+	}
+
 	return true
 }
 
 // QuorumThreshold ..
 func (v *stakedVoteWeight) QuorumThreshold() *big.Int {
-	return new(big.Int).Mul(v.totalEffectiveStakedAmount, twoThirds)
+	return v.total.Mul(twoThirds).Int
```

Comment:
.Int gives your directly the underlying raw data for the Dec. You should use .RoundInt() to get the rounded Int value.

Reply:
Actually, use Ceil() to make sure the threshold is really "more than" 2/3

Reply:
so `v.total.Mul(twoThirds).Ceil().RoundInt()` ? (because Ceil gives back dec)

Reply:
Yea, that should do it.